Compiler warning when using C23: initialization discards 'const' qualifier from pointer target type

# Bug report

### Bug description:

```c
../Python/preconfig.c: In function '_Py_get_xoption':
../Python/preconfig.c:587:24: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  587 |         wchar_t *sep = wcschr(option, L'=');
      |                        ^~~~~~
../Python/preconfig.c: In function 'preconfig_init_utf8_mode':
../Python/preconfig.c:618:24: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
{
  618 |         wchar_t *sep = wcschr(xopt, L'=');
      |                        ^~~~~~
```


### CPython versions tested on:

3.14

### Operating systems tested on:

Linux

<!-- gh-linked-prs -->
### Linked PRs
* gh-145093
* gh-145095
* gh-145101
* gh-145102
<!-- /gh-linked-prs -->

--- a/Objects/stringlib/fastsearch.h
+++ b/Objects/stringlib/fastsearch.h
@@ -69,8 +69,8 @@ STRINGLIB(find_char)(const STRINGLIB_CHAR* s, Py_ssize_t n, STRINGLIB_CHAR ch)
            and UCS4 representations. */
         if (needle != 0) {
             do {
-                void *candidate = memchr(p, needle,
-                                         (e - p) * sizeof(STRINGLIB_CHAR));
+                const void *candidate = memchr(p, needle,
+                                               (e - p) * sizeof(STRINGLIB_CHAR));
                 if (candidate == NULL)
                     return -1;
                 s1 = p;

--- a/Python/preconfig.c
+++ b/Python/preconfig.c
@@ -584,7 +584,7 @@ _Py_get_xoption(const PyWideStringList *xoptions, const wchar_t *name)
     for (Py_ssize_t i=0; i < xoptions->length; i++) {
         const wchar_t *option = xoptions->items[i];
         size_t len;
-        wchar_t *sep = wcschr(option, L'=');
+        const wchar_t *sep = wcschr(option, L'=');
         if (sep != NULL) {
             len = (sep - option);
         }
@@ -615,7 +615,7 @@ preconfig_init_utf8_mode(PyPreConfig *config, const _PyPreCmdline *cmdline)
     const wchar_t *xopt;
     xopt = _Py_get_xoption(&cmdline->xoptions, L"utf8");
     if (xopt) {
-        wchar_t *sep = wcschr(xopt, L'=');
+        const wchar_t *sep = wcschr(xopt, L'=');
         if (sep) {
             xopt = sep + 1;
             if (wcscmp(xopt, L"1") == 0) {

